mprse/mbed-os-lowlevel-tests
# Add low-level test for uart.

## Description
This is a proposition of UART low level test for Green Tea. 

This is continuation of: PR https://github.com/ARMmbed/mbed-os/pull/5273.

Requirements:

> When vendors start porting a platform to mbed OS, they need to take big steps before they actually can start running tests. One of the first requirements to run greentea is functional UART (RawSerial), unfortunately there's nothing that would tell whether the implementation is correct.
> 
>     Analize greentea to identify serial primitives that need to be tested to ensure the greentea will work correctly.
>     Create tests set:
>         it shouldn't use greentea
>         can be run on minimal (not complete) mbed OS port
>         doesn't use other part of the system
>         tests enough of UART to enable greentea
>         doesnt need to be full unit/integration tests for UART

Commit provides 2 files:
- test.py - host part of the test.  This script builds mbed part of the test and performs serial communication test.
- main.cpp - source file for mbed part of the test. This part tries to communicate with the host test using RawSerial class used by the Green Tea.

Example usage:
- copy `low_level_tests` directory into the `mbed-os` root directory.
- enter `mbed-os/low_level_tests/uart` directory,
- execute the following command: `python test.py -t GCC_ARM -m NUCLEO_F070RB -b 9600` - run test using GCC_ARM toolchain, NUCLEO_F070RB board and test serial port with baud rate set to 9600 b/s.
- `python test.py -h` - see detail usage information.


## Status
HOLD


## Migrations
 NO
